roxie/ccd/ccdfile.cpp

```diff
@@ -1418,11 +1459,63 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         {
             for (;;)
             {
-                toClose.wait(10 * 60 * 1000);  // check expired file handles every 10 minutes 
+#ifdef _CONTAINERIZED
+                unsigned checkPeriod = topology->getPropInt("@copyCheckPeriod", 60);
+#else
+                unsigned checkPeriod = 10*60;  // check expired file handles every 10 minutes, buddyCopying a little more often
+#endif
+                unsigned start = msTick();
+                toClose.wait(checkPeriod * 1000);
                 if (closing)
                     break;
-                doCloseExpired(true);
-                doCloseExpired(false);
+#ifdef _CONTAINERIZED
+                // Periodically recheck the list to see what is now local, and remove them from the buddyCopying list
+                ICopyArrayOf<ILazyFileIO> checkBuddies;
+                {
+                    CriticalBlock b(crit);
+                    if (buddyCopying.length())
+                    {
+                        buddyCopying.swapWith(checkBuddies);
+                        buddyChecking = true;
+                    }
+                }
+                if (checkBuddies.length())
+                {
+                    ForEachItemIn(idx, checkBuddies)
+                    {
+                        ILazyFileIO &check = checkBuddies.item(idx);
+                        if (traceRemoteFiles)
+                            DBGLOG("Checking whether someone has copied file %s for me", check.queryFilename());
+                        if (check.isRemote())
+                        {
+                            if (traceRemoteFiles)
+                                check.dump();
+                            if (!check.checkCopyComplete())   // Recheck whether there is a local file we can open
+                            {
+                                CriticalBlock b1(crit);
+                                buddyCopying.append(check);
+                            }
+                        }
+                    }
+                    CriticalBlock b2(crit);
+                    buddyChecking = false;
+                    if (buddyCopying.length()==0)
+                    {
+                        DBGLOG("No more data files being copied by other nodes");
+                        if (todo.ordinality()==0 && reportedFilesToCopy)
+                        {
+                            DBGLOG("No more data files to copy");
+                            reportedFilesToCopy = false;
+                        }
+                    }
+                }
+#endif
+                unsigned elapsed = msTick()-start;
+                if (elapsed >= 10*60*1000)
+                {
```

Comment:
Is it a problem, that this is unlikely to be executed in containerized?  Does start need ressetting inside this if instead of at the head of the loop?

Reply:
Yes - will fix